components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/ciba/CibaAuthRequestValidator.java

```diff
@@ -470,13 +467,47 @@ public void validateClient(String request) throws CibaAuthFailureException {
             if (log.isDebugEnabled()) {
                 log.debug("The request: " + request + " doesn't have a proper clientID.");
             }
-            throw new CibaAuthFailureException(OAuth2ErrorCodes.UNAUTHORIZED_CLIENT, "Unknown (iss) client.");
+            throw new CibaAuthFailureException(OAuth2ErrorCodes.INVALID_REQUEST, "Unknown (iss) client.");
 
         } catch (IdentityOAuth2Exception | ParseException ex) {
             throw new CibaAuthFailureException(OAuth2ErrorCodes.SERVER_ERROR, "Exception in validating for (iss). ");
         }
     }
 
+    /**
+     * Checks whether the client is valid.
+     *
+     * @param request CIBA Authentication request.
+     * @param authenticatedClient Authenticated Client ID.
+     * @throws CibaAuthFailureException CIBA Authentication Failed Exception.
+     */
+    public void validateClient(String request, String authenticatedClient) throws CibaAuthFailureException {
+        try {
+            SignedJWT signedJWT = SignedJWT.parse(request);
+            JWTClaimsSet claimsSet = signedJWT.getJWTClaimsSet();
+            String clientId = claimsSet.getIssuer();
+
+            // Validate whether the claim for the issuer is valid.
+            if (StringUtils.isBlank(claimsSet.getIssuer())) {
+                // (iss) is a blank value.
+
+                if (log.isDebugEnabled()) {
+                    log.debug("Missing issuer of the JWT of the request : " + request);
+                }
+                throw new CibaAuthFailureException(OAuth2ErrorCodes.INVALID_REQUEST, "Invalid value for (iss).");
+            }
+
+            if (!clientId.equals(authenticatedClient)) {
+                throw new CibaAuthFailureException(OAuth2ErrorCodes.INVALID_CLIENT,
+                        "Issuer does not match with the authenticated client.");
+            }
+            validateClient(request);
+
+        } catch (ParseException ex) {
+            throw new CibaAuthFailureException(OAuth2ErrorCodes.SERVER_ERROR, "Exception in validating for (iss). ");
```

Comment:
'ex' not logged or rethrown

Reply:
Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1743/commits/0d57e5040acf37b8685893a5c5cbd39f29ea9bed